### DISCUSSION TITLE
Synergy with other cockpit components

### DISCUSSION THREAD

**Discussion by leomoty** [cockpit-project/cockpit-files#650] [Ideas]
So I am curious if there is any interest in augmenting the rest of cockpit to benefit from cockpit-files, if installed it could offer some jump points from places such as storage mount points, machines and possibly others.

I've made a very simple test for my own amusement using machines:

![image](https://github.com/user-attachments/assets/3da345f8-5390-482f-9db0-3c9b55452eb1)

What I used for the example:

```jsx
function ShowFolderInFileBrowser({ text, path }) {
    if (!('files' in cockpit.manifests)) {
        return { text };
    }

    return (
        <Tooltip content={_("Open folder in File browser")}>
            <Button isInline variant="link" component="a" onClick={() => cockpit.jump(["/files#/?path=", encodeURIComponent(path)].join(""))}>
                { text }
            </Button>
        </Tooltip>
    );
}
```

**Comment by jelly** [comment]
This makes a lot of sense, we already do this for podman when a container uses a lot of memory there is a tooltip with a link to a filtered on the linked container.

**Comment by garrett** [comment]
We definitely want to do stuff like this, and have talked about it a few times already. The first goal is making the file manager complete enough, which is something we're still working on. :wink: 

(BTW: Opening the folder in files should be at the top-level, not the item itself. Or it could be a contextual menu item. It probably shouldn't be a link, as that would imply you're acting upon the file, such as opening the file or visiting a view of that file specifically.)

**Reply by leomoty** [reply]
Yea, this was just a dumb example, also it wasn't doing anything with the file, I just didnt want to pollute the example with cleaning the filename from the directory path.
